trin-core/src/types/accumulator.rs

```diff
@@ -44,6 +45,65 @@ impl MasterAccumulator {
         }
     }
 
+    pub fn latest_height(&self) -> u64 {
+        let historical_height = self.epoch_number() * EPOCH_SIZE as u64;
+        historical_height + self.header_record_count()
+    }
+
+    fn header_record_count(&self) -> u64 {
+        u64::try_from(self.current_epoch.header_records.len())
+            .expect("Header Record count should not overflow u64")
+    }
+
+    fn epoch_number(&self) -> u64 {
+        u64::try_from(self.historical_epochs.epochs.len())
+            .expect("Historical Epoch count should not overflow u64")
+    }
+
+    /// Compare that one master accumulator contains the same validation data as another,
+    /// regardless of each accumulator's current height.
+    pub fn validation_data_matches(&self, other: &Self) -> anyhow::Result<()> {
+        let historical_epochs = self
+            .historical_epochs
+            .epochs
+            .into_iter()
+            .zip(other.historical_epochs.epochs.into_iter());
+        for (index, (self_epoch, other_epoch)) in historical_epochs.into_iter().enumerate() {
+            if self_epoch != other_epoch {
```

Comment:
I guess comparing epoch by epoch here and header by header below, instead of comparing slices of epochs/headers, is to know at which epoch/header record there is a mismatch?

Reply:
Yup, pretty much. I've already found this debugging information helpful. I have a feeling that it will continue be helpful as we start introducing real accumulators into the network, passing them around, and using them to validate real data.

I guess the point you & @jacobkaufmann are making is that this design can result a lot of unnecessary work. Although I'm still not sure I understand what a "slice" is? But, I've added a check to compare the tree root hashes for the two data types before iterating. So we only perform the iteration to find the offending mismatch if the two root hashes don't match.

Reply:
Cool, that makes sense to me.

> Although I'm still not sure I understand what a "slice" is?
Array or vector of epochs.